store/store.go

```diff
@@ -437,24 +486,26 @@ func (s *Store) doRequest(client *http.Client, reqOptions *requestOptions, user
 
 	wwwAuth := resp.Header.Get("WWW-Authenticate")
 	if resp.StatusCode == 401 && user != nil && strings.Contains(wwwAuth, "needs_refresh=1") {
-		// close previous response, refresh and retry
+		// close previous response, refresh user and retry
 		resp.Body.Close()
-		err = refreshMacaroon(user)
+		err = s.refreshUser(user)
 		if err != nil {
 			return nil, err
 		}
-		if s.authContext != nil {
-			err = s.authContext.UpdateUser(user)
-			if err != nil {
-				return nil, err
-			}
-		}
-		req, err := s.newRequest(reqOptions, user)
+		// retry
+		return s.doRequest(client, reqOptions, user)
+
+	} else if resp.StatusCode == 401 && strings.Contains(wwwAuth, "refresh_device_session=1") {
```

Comment:
@matiasb actually, yes, it seems the code to achieve that wouldn't be more complicated  as far as I can tell,
top if for 401 two internal ifs for the Contains and refresh* calls?  close body at start, end in recursive do request? btw if something strange goes on we would get a long chain of recursive calls, maybe good to have a TODO about that (make this non recursive or a recursion limit)